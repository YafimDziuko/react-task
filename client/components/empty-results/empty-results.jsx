import React from 'react';
import { connect } from 'react-redux';

import { setEmptyResults, returnToMainPage } from '../../action-creators';
import './empty-results.scss';

class EmptyResults extends React.Component {
    componentWillMount() {
        this.props.setEmptyResults();
    }

    render() {
        return (
            <p className="empty-page">No films found</p>
        );
    }
} 

const mapDispatchToProps = {
    setEmptyResults
}

const mapStateToProps = (state) => {
    return {
        user: state
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(EmptyResults);