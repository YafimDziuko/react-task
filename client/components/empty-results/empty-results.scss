.empty-page {
    margin: 0;
    padding-top: 30%;
    text-align: center;
    font-size: 32px;
    font-weight: 500;
}