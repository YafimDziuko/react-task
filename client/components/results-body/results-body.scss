.results-body {
    padding-top: 15px;
    padding-bottom: 55px;
    display: flex;
    flex-wrap: wrap;

    .empty-page {
        flex: 1;
    }
}