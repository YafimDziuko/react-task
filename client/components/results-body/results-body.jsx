import React from 'react';
import ReactDOM from 'react-dom';
import { connect } from 'react-redux';
import Pagination from "react-js-pagination";
import { ResultsBar } from '../results-bar/results-bar';
import FilmItem from '../film-item/film-item';
import { EmptyResults } from '../empty-results/empty-results';
import { ResultsBodyErrorBoundary } from '../results-body-error-boundary/results-body-error-boundary';

import './results-body.scss';
import { ACTIONS } from '../../constants/app-constants';

class ResultsBody extends React.Component {
    paginationHandler(event) {
        console.log(event);
        return fetch(`http://react-cdp-api.herokuapp.com/movies?sortBy=${this.props.user.sort}&search=${this.props.user.inputValue}&searchBy=${this.props.user.search}&offset=${(event-1)*12}&limit=12`)
            .then(response => response.json())
            .then(json => this.props.paginationHandler(event, json))
    }

    render() {
        const assets = this.props.user.results;

        const children = assets.data.length ? assets.data.map(asset => (
            <FilmItem
                key={asset.id}
                id={asset.id}
                title={asset.title}
                posterUrl={asset.poster_path}
                year={Number(asset.release_date.slice(0, 4))}
                genres={asset.genres.join(', ')}
            />
        )) : <EmptyResults />;

        return (
            <ResultsBodyErrorBoundary>
            <div>
                <div className="results-body">
                    { children }               
                
                </div>
                { !!assets.data.length &&
                <div className="pagination-container">
                    <Pagination 
                        activePage={this.props.user.pagination.activePage}
                        itemsCountPerPage={12}
                        totalItemsCount={this.props.user.count}
                        pageRangeDisplayed={5}
                        onChange={(event)=>this.paginationHandler(event)}
                    /> 
                </div>
                }
            </div>
            </ResultsBodyErrorBoundary>
        );
    }
};

const mapStateToProps = (state) => {
    return {
        user: state
    };
};


const mapDispatchToProps = (dispatch) => {
    return {
        paginationHandler: (activePage, data) => {
            dispatch({
                type: ACTIONS.PAGINATION_HANDLER,
                activePage: activePage,
                results: data
            })
        }
    };
};


export default connect(mapStateToProps, mapDispatchToProps)(ResultsBody);