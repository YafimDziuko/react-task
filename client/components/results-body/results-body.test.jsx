import React from 'react';
import { ResultsBody } from './results-body';
import { EmptyResults } from '../empty-results/empty-results';

const resultAssets = [
    { title: 'Asset 1' },
    { title: 'Asset 2' },
    { title: 'Asset 3' },
    { title: 'Asset 4' },
    { title: 'Asset 5' },
    { title: 'Asset 6' },
    { title: 'Asset 7' },
    { title: 'Asset 8' },
    { title: 'Asset 9' },
];

describe('<ResultsBody/>', function() {
  it('should render ResultsBody and match snapshot', function() {
      const wrapper = shallow(<ResultsBody assets={resultAssets}/>);

      expect(wrapper).toMatchSnapshot();
  });

  it('should render ResultsBody and match snapshot', function() {
    const wrapper = shallow(<ResultsBody assets={[]}/>);

    expect(wrapper.contains(<EmptyResults/>)).toEqual(true);
    expect(wrapper).toMatchSnapshot();
  });
});