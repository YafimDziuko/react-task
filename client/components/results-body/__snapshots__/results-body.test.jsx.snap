// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultsBody/> should render ResultsBody and match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResultsBody
    dispatch={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
    user={
      Object {
        "fullItem": Object {
          "isActive": false,
        },
        "inputValue": "",
        "results": Object {
          "data": Array [
            Object {
              "genres": Array [
                "genre",
              ],
              "id": 123,
              "key": 123,
              "posterUrl": "url",
              "release_date": "2018-12-01",
              "title": "title",
            },
          ],
        },
        "resultsCount": -1,
        "search": "title",
        "searchInfo": Object {},
        "sort": "releaseDate",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div>
        <div
          className="results-body"
        >
          <Connect(FilmItem)
            genres="genre"
            id={123}
            posterUrl={undefined}
            title="title"
            year={2018}
          />
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="results-body"
        >
          <Connect(FilmItem)
            genres="genre"
            id={123}
            posterUrl={undefined}
            title="title"
            year={2018}
          />
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(FilmItem)
              genres="genre"
              id={123}
              posterUrl={undefined}
              title="title"
              year={2018}
            />,
          ],
          "className": "results-body",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "123",
            "nodeType": "class",
            "props": Object {
              "genres": "genre",
              "id": 123,
              "posterUrl": undefined,
              "title": "title",
              "year": 2018,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div>
          <div
            className="results-body"
          >
            <Connect(FilmItem)
              genres="genre"
              id={123}
              posterUrl={undefined}
              title="title"
              year={2018}
            />
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="results-body"
          >
            <Connect(FilmItem)
              genres="genre"
              id={123}
              posterUrl={undefined}
              title="title"
              year={2018}
            />
          </div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(FilmItem)
                genres="genre"
                id={123}
                posterUrl={undefined}
                title="title"
                year={2018}
              />,
            ],
            "className": "results-body",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "123",
              "nodeType": "class",
              "props": Object {
                "genres": "genre",
                "id": 123,
                "posterUrl": undefined,
                "title": "title",
                "year": 2018,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ResultsBody/> should render ResultsBody and match snapshot 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResultsBody
    dispatch={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
    user={
      Object {
        "fullItem": Object {
          "isActive": false,
        },
        "inputValue": "",
        "results": Object {
          "data": Array [],
        },
        "resultsCount": -1,
        "search": "title",
        "searchInfo": Object {},
        "sort": "releaseDate",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div>
        <div
          className="results-body"
        >
          <Connect(EmptyResults) />
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="results-body"
        >
          <Connect(EmptyResults) />
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(EmptyResults) />,
          "className": "results-body",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div>
          <div
            className="results-body"
          >
            <Connect(EmptyResults) />
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="results-body"
          >
            <Connect(EmptyResults) />
          </div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(EmptyResults) />,
            "className": "results-body",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
